Runtime/Services/BacktraceSession.cs

```diff
@@ -0,0 +1,317 @@
+                }
+            });
+        }
+
+        /// <summary>
+        /// Add unique event to next Backtrace session request
+        /// </summary>
+        /// <param name="attributeName">attribute name</param>
+        public bool AddUniqueEventAttribute(string attributeName, IDictionary<string, string> attributes = null)
+        {
+            if (!ShouldProcessEvent(attributeName))
+            {
+                Debug.LogWarning("Skipping report: Reached store limit or event has empty name.");
+                return false;
+            }
+            // add to event attributes, attribute provider attributes
+            if (attributes == null)
+            {
+                attributes = new Dictionary<string, string>();
+            }
+            _attributeProvider.AddAttributes(attributes);
+
+            // validate if unique event attribute is available and
+            // prevent undefined attributes
+            if (attributes.TryGetValue(attributeName, out string attributeValue) == false || string.IsNullOrEmpty(attributeValue))
+            {
+                Debug.LogWarning("Attribute name is not available in attribute scope. Please define attribute to set unique event.");
+                return false;
+            }
+            var existingEvent = UniqueEvents.FirstOrDefault(m => m.Name == attributeName);
+            if (existingEvent != null)
+            {
+                UniqueEvents.Remove(existingEvent);
+            }
+
+            var @event = new UniqueEvent(attributeName, DateTimeHelper.Timestamp(), attributes);
+            UniqueEvents.AddLast(@event);
+            return true;
+        }
+
+        /// <summary>
+        /// Determine if Backtrace Session can add next event to store
+        /// </summary>
+        /// <param name="name">event name</param>
+        /// <returns>True if we're able to add. Otherwise false.</returns>
+        private bool ShouldProcessEvent(string name)
+        {
+            return !string.IsNullOrEmpty(name) && (_maximumNumberOfEventsInStore == 0 || (Count() + 1 <= _maximumNumberOfEventsInStore));
+        }
+
+
+        /// <summary>
+        /// Get number of events available right now in store
+        /// </summary>
+        /// <returns>number of events in store</returns>
+        public int Count()
+        {
+            return UniqueEvents.Count + SessionEvents.Count;
+        }
+
+        /// <summary>
+        /// Add session event to next Backtrace session request
+        /// </summary>
+        /// <param name="eventName">session event name</param>
+        public bool AddSessionEvent(string eventName, IDictionary<string, string> attributes = null)
+        {
+            if (!ShouldProcessEvent(eventName))
+            {
+                Debug.LogWarning("Skipping report: Reached store limit or event has empty name.");
+                return false;
+            }
+            var existingEvent = SessionEvents.FirstOrDefault(m => m.Name == eventName);

```

Comment:
We need to allow sending duplicate events since session events is a counter.

Reply:
Thanks. Removed this check. 

Let me also clarify - if we have two the same session events, maybe we should use some kind of counter instead of putting two the same events in session events array?

Reply:
Maybe, but currently the service isn't set up for it and telling when two events are different is ust involved enough that I'd rather not.  I would expect that the number of events from an individual client is going to be low in practice.